diff a/src/jdk.management.jfr/share/classes/jdk/management/jfr/DiskRepository.java b/src/jdk.management.jfr/share/classes/jdk/management/jfr/DiskRepository.java
--- a/src/jdk.management.jfr/share/classes/jdk/management/jfr/DiskRepository.java
+++ b/src/jdk.management.jfr/share/classes/jdk/management/jfr/DiskRepository.java
@@ -1,7 +1,7 @@
 /*
- * Copyright (c) 2020, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, 2025, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License version 2 only, as
  * published by the Free Software Foundation.  Oracle designates this
@@ -149,11 +149,11 @@
     private long size;
 
     public DiskRepository(Path path, boolean deleteDirectory) throws IOException {
         this.directory = path;
         this.deleteDirectory = deleteDirectory;
-        this.chunkFilename = ChunkFilename.newUnpriviliged(path);
+        this.chunkFilename = new ChunkFilename(path);
     }
 
     public synchronized void write(byte[] bytes) throws IOException {
         barrier.check();
         index = 0;
