--- /home/sherman/TL/zlib-1.2.3_ORG/crc32.c	Sun Jun 12 16:56:07 2005
+++ zcrc32.c	Tue Aug 25 14:22:41 2009
@@ -216,8 +216,8 @@
 #define DO8 DO1; DO1; DO1; DO1; DO1; DO1; DO1; DO1
 
 /* ========================================================================= */
-unsigned long ZEXPORT crc32(crc, buf, len)
-    unsigned long crc;
+uLong ZEXPORT crc32(crc, buf, len)
+    uLong crc;
     const unsigned char FAR *buf;
     unsigned len;
 {
@@ -234,9 +234,9 @@
 
         endian = 1;
         if (*((unsigned char *)(&endian)))
-            return crc32_little(crc, buf, len);
+            return (uLong)crc32_little(crc, buf, len);
         else
-            return crc32_big(crc, buf, len);
+            return (uLong)crc32_big(crc, buf, len);
     }
 #endif /* BYFOUR */
     crc = crc ^ 0xffffffffUL;
