--- /home/sherman/TL/zlib-1.2.3_ORG/zconf.h	Fri May 27 23:40:35 2005
+++ zconf.h	Tue Aug 25 14:22:28 2009
@@ -8,6 +8,9 @@
 #ifndef ZCONF_H
 #define ZCONF_H
 
+/* for _LP64 */
+#include <sys/types.h>
+
 /*
  * If you *really* need a unique prefix for all types and library functions,
  * compile with -DZ_PREFIX. The "standard" zlib should be compiled without it.
@@ -261,7 +264,11 @@
 typedef unsigned char  Byte;  /* 8 bits */
 #endif
 typedef unsigned int   uInt;  /* 16 bits or more */
+#ifdef _LP64
+typedef unsigned int  uLong;  /* 32 bits or more */
+#else
 typedef unsigned long  uLong; /* 32 bits or more */
+#endif
 
 #ifdef SMALL_MEDIUM
    /* Borland C/C++ and some old MSC versions ignore FAR inside typedef */
