diff a/test/jdk/jdk/jfr/jvm/MyCommitRegisteredFalseEvent.java b/test/jdk/jdk/jfr/jvm/MyCommitRegisteredFalseEvent.java
--- a/test/jdk/jdk/jfr/jvm/MyCommitRegisteredFalseEvent.java
+++ b/test/jdk/jdk/jfr/jvm/MyCommitRegisteredFalseEvent.java
@@ -1,7 +1,7 @@
 /*
- * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2022, 2025, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License version 2 only, as
  * published by the Free Software Foundation.
@@ -26,11 +26,11 @@
 
 // Class used by TestGetEventWriter
 @Registered(false)
 public class MyCommitRegisteredFalseEvent extends E implements Runnable {
     public void myCommit() {
-        PlaceholderEventWriterFactory.getEventWriter(4711L);
+        PlaceholderEventWriter.getEventWriter();
         throw new RuntimeException("Should not reach here");
     }
 
     @Override
     public void run() {
