diff a/test/jdk/jdk/jfr/jvm/NonEvent.java b/test/jdk/jdk/jfr/jvm/NonEvent.java
--- a/test/jdk/jdk/jfr/jvm/NonEvent.java
+++ b/test/jdk/jdk/jfr/jvm/NonEvent.java
@@ -1,7 +1,7 @@
 /*
- * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2022, 2025, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License version 2 only, as
  * published by the Free Software Foundation.
@@ -23,11 +23,11 @@
 package jdk.jfr.jvm;
 
 // Class used by TestGetEventWriter
 public class NonEvent implements Runnable {
     public void commit() {
-        PlaceholderEventWriter ew = PlaceholderEventWriterFactory.getEventWriter(4711L);
+        PlaceholderEventWriter ew = PlaceholderEventWriter.getEventWriter();;
         throw new RuntimeException("Should not reach here " + ew);
     }
 
     @Override
     public void run() {
