diff a/src/jdk.jfr/share/classes/jdk/jfr/SettingControl.java b/src/jdk.jfr/share/classes/jdk/jfr/SettingControl.java
--- a/src/jdk.jfr/share/classes/jdk/jfr/SettingControl.java
+++ b/src/jdk.jfr/share/classes/jdk/jfr/SettingControl.java
@@ -1,7 +1,7 @@
 /*
- * Copyright (c) 2016, 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2016, 2025, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License version 2 only, as
  * published by the Free Software Foundation.  Oracle designates this
@@ -23,16 +23,12 @@
  * questions.
  */
 
 package jdk.jfr;
 
-import java.security.AccessControlContext;
-import java.security.AccessController;
 import java.util.Set;
 
-import jdk.jfr.internal.settings.JDKSettingControl;
-
 /**
  * Base class to extend to create setting controls.
  * <p>
  * The following example shows a naive implementation of a setting control for
  * regular expressions:
@@ -75,34 +71,14 @@
  * @since 9
  */
 @MetadataDefinition
 public abstract class SettingControl {
 
-    @SuppressWarnings("removal")
-    private final AccessControlContext context;
-    private final boolean initialized;
-
     /**
      * Constructor for invocation by subclass constructors.
      */
-    @SuppressWarnings("removal")
     protected SettingControl() {
-        context = this instanceof JDKSettingControl ? null : AccessController.getContext();
-        initialized = true;
-    }
-
-    @SuppressWarnings("removal")
-    final AccessControlContext getContext() {
-        // Ensure object state is safe
-        if (!initialized) {
-            throw new InternalError("Object must be initialized before security context can be retrieved");
-        }
-        AccessControlContext c = this.context;
-        if (c == null && !(this instanceof JDKSettingControl)) {
-            throw new InternalError("Security context can only be null for trusted setting controls");
-        }
-        return c;
     }
 
     /**
      * Combines the setting values for all running recordings into one value when
      * multiple recordings are running at the same time,
