diff --git a/make/devkit/createWindowsDevkit2019.sh b/make/devkit/createWindowsDevkit2019.sh
index a5c89f09faa..b190118fce4 100644
--- a/make/devkit/createWindowsDevkit2019.sh
+++ b/make/devkit/createWindowsDevkit2019.sh
@@ -109,9 +109,11 @@ MSVCP_DLL=${MSVC_DIR}/msvcp${VS_DLL_VERSION}.dll
 # Copy Visual Studio files
 
 TOOLS_VERSION="$(ls "$VS_INSTALL_DIR/VC/Tools/MSVC" | sort -r -n | head -n1)"
+TOOLS_VERSION=14.21.27702
 echo "Found Tools version: $TOOLS_VERSION"
 VC_SUBDIR="VC/Tools/MSVC/$TOOLS_VERSION"
 REDIST_VERSION="$(ls "$VS_INSTALL_DIR/VC/Redist/MSVC" | sort -r -n | head -n1)"
+REDIST_VERSION=14.21.27702
 echo "Found Redist version: $REDIST_VERSION"
 REDIST_SUBDIR="VC/Redist/MSVC/$REDIST_VERSION"
 echo "Copying VC..."
