diff a/src/jdk.jfr/share/classes/jdk/jfr/internal/TypeLibrary.java b/src/jdk.jfr/share/classes/jdk/jfr/internal/TypeLibrary.java
--- a/src/jdk.jfr/share/classes/jdk/jfr/internal/TypeLibrary.java
+++ b/src/jdk.jfr/share/classes/jdk/jfr/internal/TypeLibrary.java
@@ -1,7 +1,7 @@
 /*
- * Copyright (c) 2016, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2016, 2025, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License version 2 only, as
  * published by the Free Software Foundation.  Oracle designates this
@@ -187,11 +187,11 @@
                 String proxyPackage = annotation.getClass().getPackageName();
                 Module jfrModule = TypeLibrary.class.getModule();
                 Modules.addExports(proxyModule, proxyPackage, jfrModule);
             }
         }
-        SecuritySupport.setAccessible(m);
+        m.setAccessible(true);
         try {
             return m.invoke(annotation, new Object[0]);
         } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
             throw (Error) new InternalError("Could not get value for method " + methodName + " in annotation " + annotation.getClass().getName());
         }
