diff a/test/jdk/jdk/jfr/jcmd/TestJcmdConfigure.java b/test/jdk/jdk/jfr/jcmd/TestJcmdConfigure.java
--- a/test/jdk/jdk/jfr/jcmd/TestJcmdConfigure.java
+++ b/test/jdk/jdk/jfr/jcmd/TestJcmdConfigure.java
@@ -1,7 +1,7 @@
 /*
- * Copyright (c) 2015, 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2015, 2025, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License version 2 only, as
  * published by the Free Software Foundation.
@@ -23,16 +23,16 @@
 
 package jdk.jfr.jcmd;
 
 import java.io.File;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
 
 import jdk.jfr.internal.Repository;
-import jdk.jfr.internal.SecuritySupport.SafePath;
 import jdk.jfr.internal.Options;
 import jdk.test.lib.Asserts;
 import jdk.test.lib.Utils;
 
 /**
@@ -129,21 +129,21 @@
     private static void testRepository(){
         final String findWhat = "[info][jfr] Same base repository path " + REPOSITORYPATH_1 + " is set";
 
         try {
             JcmdHelper.jcmd("JFR.configure", REPOSITORYPATH_SETTING_1);
-            SafePath initialPath = Repository.getRepository().getRepositoryPath();
+            Path initialPath = Repository.getRepository().getRepositoryPath();
 
             JcmdHelper.jcmd("JFR.configure", REPOSITORYPATH_SETTING_1);
-            SafePath samePath = Repository.getRepository().getRepositoryPath();
+            Path samePath = Repository.getRepository().getRepositoryPath();
             Asserts.assertTrue(samePath.equals(initialPath));
 
             List<String> lines = Files.readAllLines(Paths.get(JFR_UNIFIED_LOG_FILE));
             Asserts.assertTrue(lines.stream().anyMatch(l->l.contains(findWhat)));
 
             JcmdHelper.jcmd("JFR.configure", REPOSITORYPATH_SETTING_2);
-            SafePath changedPath = Repository.getRepository().getRepositoryPath();
+            Path changedPath = Repository.getRepository().getRepositoryPath();
 
             Asserts.assertFalse(changedPath.equals(initialPath));
 
         } catch(Exception e) {
             testExceptions.add(e);
