diff a/src/jdk.jfr/share/classes/jdk/jfr/internal/settings/LevelSetting.java b/src/jdk.jfr/share/classes/jdk/jfr/internal/settings/LevelSetting.java
--- a/src/jdk.jfr/share/classes/jdk/jfr/internal/settings/LevelSetting.java
+++ b/src/jdk.jfr/share/classes/jdk/jfr/internal/settings/LevelSetting.java
@@ -1,7 +1,7 @@
 /*
- * Copyright (c) 2023, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023, 2025, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License version 2 only, as
  * published by the Free Software Foundation.  Oracle designates this
@@ -31,17 +31,18 @@
 import java.util.Set;
 
 import jdk.jfr.Label;
 import jdk.jfr.MetadataDefinition;
 import jdk.jfr.Name;
+import jdk.jfr.SettingControl;
 import jdk.jfr.internal.PlatformEventType;
 import jdk.jfr.internal.Type;
 
 @MetadataDefinition
 @Label("Level")
 @Name(Type.SETTINGS_PREFIX + "Level")
-public final class LevelSetting extends JDKSettingControl {
+public final class LevelSetting extends SettingControl {
     private final PlatformEventType eventType;
     private final List<String> levels;
     private String value;
 
     public LevelSetting(PlatformEventType eventType, String[] levels) {
